### PR TITLE
Memory leak cleanup

### LINKED ISSUE #137: Memory leak in RK4
We don't run RK4 very often. Running EC60to30 on 576 cores on grizzly with gnu optimized, it runs for 7 days and then dies with memory errors.

### PR DESCRIPTION
**Pull request by mark-petersen**
- Add needed deallocation lines.  
- Remove unused variable declarations and allocations.

### CODE REVIEW

**Comment by mark-petersen** [comment]
@sbrus89 this is a small clean-up follow-on to issue #137. I think a visual inspection and sanity check is sufficient.

File: src/core_ocean/shared/mpas_ocn_tracer_hmix_redi.F [hunk 1]
```diff
@@ -143,7 +143,7 @@ subroutine ocn_tracer_hmix_redi_tend(meshPool, scratchPool, layerThicknessEdge,
          dTracerdZTopOfCell, dTracerdZTopOfEdge, areaCellSum
```
[the remark this answers is not in the document]
Comment (mark-petersen) [reply src/core_ocean/shared/mpas_ocn_tracer_hmix_redi.F]: Thanks for checking so carefully. I looked again, and that variable should stay, but was missing a deallocate.  Changing now...

File: src/core_ocean/shared/mpas_ocn_tracer_hmix_redi.F [hunk 2]
```diff
@@ -143,7 +143,7 @@ subroutine ocn_tracer_hmix_redi_tend(meshPool, scratchPool, layerThicknessEdge,
...
 
       type (field2DReal), pointer :: gradTracerEdgeField, gradTracerTopOfEdgeField, gradHTracerSlopedTopOfCellField, &
-                                     dTracerdZTopOfCellField, dTracerdZTopOfEdgeField, areaCellSumField
```
Comment (xylar) [inline src/core_ocean/shared/mpas_ocn_tracer_hmix_redi.F]: @mark-petersen, this change needs to be reverted.  Maybe this is what your comment above refers to, but `areaCellSumField` is needed.  The current code never allocates `areaCellSum` but assigns to it.  Bad news!

File: src/core_ocean/shared/mpas_ocn_tracer_hmix_redi.F [hunk 3]
```diff
@@ -182,22 +182,19 @@ subroutine ocn_tracer_hmix_redi_tend(meshPool, scratchPool, layerThicknessEdge,
       call mpas_pool_get_field(scratchPool, 'gradHTracerSlopedTopOfCell', gradHTracerSlopedTopOfCellField)
       call mpas_pool_get_field(scratchPool, 'dTracerdZTopOfCell', dTracerdZTopOfCellField)
       call mpas_pool_get_field(scratchPool, 'dTracerdZTopOfEdge', dTracerdZTopOfEdgeField)
-      call mpas_pool_get_field(scratchPool, 'areaCellSum', areaCellSumField)
```
Comment (xylar) [inline src/core_ocean/shared/mpas_ocn_tracer_hmix_redi.F]: @mark-petersen, this change needs to be reverted.

File: src/core_ocean/shared/mpas_ocn_tracer_hmix_redi.F [hunk 4]
```diff
@@ -182,22 +182,19 @@ subroutine ocn_tracer_hmix_redi_tend(meshPool, scratchPool, layerThicknessEdge,
...
 
       call mpas_allocate_scratch_field(gradTracerEdgeField, .true., .false.)
       call mpas_allocate_scratch_field(gradTracerTopOfEdgeField, .true., .false.)
       call mpas_allocate_scratch_field(gradHTracerSlopedTopOfCellField, .true., .false.)
       call mpas_allocate_scratch_field(dTracerdZTopOfCellField, .true., .false.)
       call mpas_allocate_scratch_field(dTracerdZTopOfEdgeField, .true., .false.)
-      call mpas_allocate_scratch_field(areaCellSumField, .true., .false.)
```
Comment (xylar) [inline src/core_ocean/shared/mpas_ocn_tracer_hmix_redi.F]: @mark-petersen, this change needs to be reverted.

File: src/core_ocean/shared/mpas_ocn_tracer_hmix_redi.F [hunk 5]
```diff
@@ -182,22 +182,19 @@ subroutine ocn_tracer_hmix_redi_tend(meshPool, scratchPool, layerThicknessEdge,
...
       call mpas_threading_barrier()
 
       gradTracerEdge => gradTracerEdgeField % array
       gradTracerTopOfEdge => gradTracerTopOfEdgeField % array
       gradHTracerSlopedTopOfCell => gradHTracerSlopedTopOfCellField % array
       dTracerdZTopOfCell => dTracerdZTopOfCellField % array
       dTracerdZTopOfEdge => dTracerdZTopOfEdgeField % array
-      areaCellSum => areaCellSumField % array
```
[the remark this answers is not in the document]
Comment (xylar) [reply src/core_ocean/shared/mpas_ocn_tracer_hmix_redi.F]: @mark-petersen, this change needs to be reverted.

**Comment by mark-petersen** [comment]
@xylar Thanks for checking. I made those fixes locally last week as well, then didn't get a chance to finish and push the commits back up. I'll button this one up in the next week.

**Comment by mark-petersen** [comment]
Revised based on comments above. Rebased, tested with both optimized and debug, intel and gnu. All tested nightly regression suite. This PR is a similar topic to all the performance and threading PRs, so I'll merge in with the upcoming E3SM PR.

**Comment by mark-petersen** [comment]
Just a note, a convenient way to find missing deallocate lines is:
```
git grep -h mpas_allocate_scratch_field>alloctmp.F
git grep -h mpas_deallocate_scratch_field>dealloctmp.F
vimdiff alloctmp.F dealloctmp.F
```
because it shows a line-for-line match between the allocate and deallocate lines, like this:
image

**Comment by mark-petersen** [comment]
Merging in. Requested changes from first round of reviews were made, and this is needed for an E3SM PR.

**Pull request merged**